### DISCUSSION TITLE
Loading zsh-edit

### DISCUSSION THREAD

**Discussion by ittayd** [zdharma-continuum/zinit#631] [Q&A]
added `zinit light marlonrichert/zsh-edit` (github link: https://github.com/marlonrichert/zsh-edit), I get these errors:
```
(anon):10: command not found: zsh-edit
(anon):unfunction:12: no such hash table element: zsh-edit
```

**Comment by ittayd** [comment]
@NICHOLAS85 , perhaps you can help? I saw you load zsh-edit in your zshrc, but copy-pasting the code there didn't work for me (maybe the patch is required?)

**Comment by vladdoster** [comment]
@ittayd,

```zsh
zinit for \
    as'program' \
    compile'functions/*~*.zwc' \
    pick'functions/bind' \
  @marlonrichert/zsh-edit
```

**Reply by ittayd** [reply]
Thanks. No errors, but can you please explain what this does? In particular, I saw that zsh-edit.plugin.zsh runs zsh-edit.

**Reply by ittayd** [reply]
Seems like the key bindings that are supposed to be added are not . In particular `Backward shell word`

**Comment by ittayd** [comment]
zinit report shows:
```
zinit report marlonrichert/zsh-edit                                                                                                                                          ✔
Report for marlonrichert/zsh-edit plugin
----------------------------------------
Source zsh-edit.plugin.zsh (reporting enabled)
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/bind with options -U -z
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/zsh-edit with options -U -z
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/.edit.execute-cmd with options -U -z
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/.edit.find-replace with options -U -z
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/.edit.insert-word with options -U -z
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/.edit.move-word with options -U -z
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/_edit._show_clipboard with options -U -z
Autoload /home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit/functions/.edit.visual-yank with options -U -z

Functions created:
bind

FPATH elements added:
/home/ittay/.local/share/zinit/plugins/marlonrichert---zsh-edit

Variables added or redefined:
_edit_opts  [                  "" -> array-hideval              ]
fpath       [  array-tied-special -> array-tied-unique-special  ]
FPATH       [ scalar-tied-special -> scalar-tied-unique-special ]

Completions:
_bind [enabled]
_edit._show_clipboard [enabled]
```

if in `zsh-edit.plugin.zsh` I add the line `  autoload -Uz $fdir/zsh-edit` after `  autoload -Uz $fdir/{bind,zsh-edit} $fdir/[._]edit.*~*.zwc(DN)`, then the report shows zsh-edit has been configured and all works well

**Reply by ittayd** [reply]
So trying to use on of the key bindings the plugin does shows `No such shell function `.edit.move-word'`.

**Comment by ittayd** [comment]
BTW, `zinit light marlonrichert/zsh-edit` doesn't remove reporting. I tried to use it thinking that maybe the overload of `autoload` is the culprit.

**Reply by ittayd** [reply]
If I add `which autoload` inside zsh-edit.plugin.zsh, it shows autoload is a function, even though I use `light`

**Comment by ittayd** [comment]
Note the documentation states there's a function [.zinit-shadow-off](https://zdharma-continuum.github.io/zinit/wiki/zinit.zsh/#zinit-shadow-off) but it seems the function is .zinit-tmp-subst-off. Calling it before loading the plugin doesn't prevent autoload from being shadowed
